hail/python/hailtop/aiogoogle/client/compute_client.py

```diff
@@ -1,29 +1,33 @@
+import copy
 import uuid
-from typing import Mapping, Any, Optional, MutableMapping
+import asyncio
+from typing import Mapping, Any, Optional, MutableMapping, List, Dict
 import logging
 
 from .base_client import BaseClient
-from hailtop.utils import sleep_and_backoff
+from hailtop.utils import retry_transient_errors
 
 log = logging.getLogger('compute_client')
 
 
-async def request_with_wait_for_done(request_f, path, params: MutableMapping[str, Any] = None, **kwargs):
-    assert 'params' not in kwargs
+class GCPError:
+    def __init__(self, code, message):
+        self.code = code
+        self.message = message
 
-    if params is None:
-        params = {}
 
-    request_uuid = str(uuid.uuid4())
-    if 'requestId' not in params:
-        params['requestId'] = request_uuid
+class GCPOperationError(Exception):
+    def __init__(self, status: int, message: str, errors: Optional[List[GCPError]], response: Dict[str, Any]):
+        super().__init__(message)
+        self.status = status
+        self.message = message
+        self.errors = errors or []
```

Comment:
I don't think we need to keep the errors around, we have the codes and messages, right? In that case we probably don't need to have GCPError at all, since we're storing the information split.